core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +111,53 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory in the scm
+     * @param pluginName the name of the plugin
+     * @param scm the valid scm link
+     * @return folderPath from the scm
+     */
+    private static String searchPomFiles(File directory, String pluginName, String scm) {
+        File[] files = directory.listFiles();
+        if (files == null) {
+            return scm;
+        }
```

Comment:
This should be an error case right?

You are checking for files at the root of the SCM, and there are none, not even a `pom.xml` file?

Reply:
Here I  first check if the file is a directory, and then pass it here. So if it a directory then it should list files that it contains.

I only look for pom files in the directory and not exactly at the root of the SCM.